oximeter/oximeter/src/histogram.rs

```diff
@@ -508,66 +704,136 @@ where
         let mut counts = Vec::with_capacity(self.n_bins());
-
-        // The first bin may either be BinRange::To or BinRange::Range.
         for bin in self.bins.iter() {
             match bin.range {
                 BinRange::Range { start, .. } => {
                     bins.push(start);
-                },
-                BinRange::RangeFrom{start} => {
+                }
+                BinRange::RangeFrom { start} => {
                     bins.push(start);
-                },
+                }
                 _ => unreachable!("No bins in a constructed histogram should be of type RangeTo"),
             }
             counts.push(bin.count);
         }
         (bins, counts)
     }
 
-    /// Construct a histogram from a start time and paired arrays with the left bin-edge and counts.
-    pub fn from_arrays(
-        start_time: DateTime<Utc>,
-        bins: Vec<T>,
-        counts: Vec<u64>,
-    ) -> Result<Self, HistogramError> {
-        if bins.len() != counts.len() {
-            return Err(HistogramError::ArraySizeMismatch {
-                n_bins: bins.len(),
-                n_counts: counts.len(),
-            });
+    /// Return the minimum value of inputs to the histogram.
+    pub fn min(&self) -> T {
+        self.min
+    }
+
+    /// Return the maximum value of all inputs to the histogram.
+    pub fn max(&self) -> T {
+        self.max
+    }
+
+    /// Return the mean of all inputs/samples in the histogram.
+    pub fn mean(&self) -> Option<f64> {
+        if self.n_samples() > 0 {
+            self.sum_of_samples
+                .to_f64()
+                .map(|sum| sum / (self.n_samples() as f64))
+        } else {
+            None
         }
-        let mut hist = Self::new(&bins)?;
-        hist.start_time = start_time;
-        let mut n_samples = 0;
-        for (bin, count) in hist.bins.iter_mut().zip(counts.into_iter()) {
-            bin.count = count;
-            n_samples += count;
+    }
+
+    /// Return the sum of all inputs to the histogram.
+    pub fn sum_of_samples(&self) -> T::Width {
+        self.sum_of_samples
+    }
+
+    /// Return the sum of the squares of all inputs to the histogram.
+    pub fn sum_of_squares(&self) -> T::Width {
+        self.sum_of_squares
+    }
+
+    /// Return the standard deviation for inputs to the histogram.
+    pub fn std_dev(&self) -> Option<f64> {
+        if self.n_samples() > 1 {
+            self.mean().and_then(|mean| {
+                let sum_of_squares = self.sum_of_squares.to_f64()?;
+                let sum_of_samples = self.sum_of_samples.to_f64()?;
+
+                let variance = (sum_of_squares - (sum_of_samples * mean))
+                    / (self.n_samples() as f64);
+                Some(variance.sqrt())
+            })
+        } else {
+            None
+        }
+    }
+
+    /// Return the sample standard deviation for inputs to the histogram.
+    pub fn sample_std_dev(&self) -> Option<f64> {
```

Comment:
I think we might want to improve the documentation of this method and the `std_dev()` method. `std_dev()` actually returns the standard deviation of the samples seen by the histogram. This is a biased estimator of the standard deviation of the population underlying those samples. The value returned from `sample_std_dev()` is technically the _corrected sample standard deviation_, though we do often call it just the "sample standard deviation".

I'd recommend making clear that `std_dev()` is a biased estimate of the population deviation, and that `sample_std_dev()` is an unbiased estimate applying Bessel's correction. I might also suggest we make a recommendation that people usually call `sample_std_dev`, as that's more often what people want: an unbiased estimator of the deviation in the underlying distribution.